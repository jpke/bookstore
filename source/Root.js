import React from 'react';
import {connect} from 'react-redux';
import { StyleSheet,
         Text,
         TouchableHighlight,
         View,
         Navigator,
         Image
       } from 'react-native';
import HomeScreen from './HomeScreen';
import Admin from './Admin';
import Order from './Order';
import OrderHistory from './OrderHistory';
import UserInfo from './UserInfo';
import AddressInfo from './AddressInfo';
import PaymentInfo from './PaymentInfo';
import ReviewOrder from './ReviewOrder';


class Root extends React.Component {
  constructor() {
    super();
    this.state = {
      userInfo: {
        firstName: "",
        lastName: "",
        email: ""
      },
      deliveryAddress: {
        lineOne: "",
        lineTwo: "",
        city: "",
        state: "",
        zip: ""
      },
      useDeliveryAddressAsBilling: false,
      billingAddress: {
        lineOne: "",
        lineTwo: "",
        city: "",
        state: "",
        zip: ""
      },
      creditCardInfo: {
        nameOnCard: "",
        cardNumber: "",
        securityCode: ""
      },
      availableBooks: [
        {
          title: "The Evolution of Civilizations",
          author: "Carroll Quigley",
        },
        {
          title: "How the Mind Works",
          author: "Stephan Pinker",

        }
      ],
      orders: [
        {
          title: "The Lexus and the Olive Tree",
          author: "Thomas Friedman",
          date: new Date()
        },
        {
          title: "Wit and Wisdom",
          author: "Toby Reynolds",
          date: new Date(Date.now() - (1000*60*60*24))
        }
      ],
      bookToOrder: "",
      newBook: {
        title: "",
        author: ""
      }
    }
  }

  selectNewBook(newBook) {
    this.setState({
      bookToOrder: newBook
    })
  }

  submitInfo(info, type, option) {
    if(type === 'AdminAddBook') {
      if(info.title === "" || info.author === "") return;
      let books = this.state.availableBooks.concat(info);
      this.setState({
        availableBooks: books
      });
    }
    else if(type === 'UserInfo') {
      this.setState({
        userInfo: info
      });
    }
    else if(type === 'Delivery') {
      //use delivery address as billing address?
      option ?
        this.setState({
          deliveryAddress: info,
          billingAddress: info,
          useDeliveryAddressAsBilling: true
        })
      :
        this.setState({
          deliveryAddress: info,
          useDeliveryAddressAsBilling: false
        })
    }
    else if(type === 'Billing') {
      this.setState({
        billingAddress: info
      });
    }
    else if(type === 'PaymentInfo') {
      this.setState({
        creditCardInfo: info
      });
    }
    else if(type === 'SubmitOrder') {
      info.date = new Date();
      let orders = this.state.orders.concat(info);
      let books = this.state.availableBooks.slice();
      books = books.filter(book => book.title !== info.title)
      this.setState({
        orders: orders,
        availableBooks: books
      });
    }
  }

  renderScene(route, navigator) {
   if(route.name == 'HomeScreen') {
     return <HomeScreen navigator={navigator} />
   }
   if(route.name == 'Admin') {
     return <Admin navigator={navigator} />
   }
   if(route.name == 'Order') {
     return <Order navigator={navigator} />
   }
   if(route.name == 'OrderHistory') {
     return <OrderHistory navigator={navigator} />
   }
   if(route.name == 'UserInfo') {
     return <UserInfo navigator={navigator} />
   }
   if(route.name == 'AddressInfo') {
     return <AddressInfo navigator={navigator}
            addressType={route.option === 'delivery' ? 'Delivery' : 'Billing'}
            address={route.option === 'delivery' ? this.props.deliveryAddress : this.props.billingAddress}
            />
   }
   if(route.name == 'PaymentInfo') {
     return <PaymentInfo navigator={navigator}
            creditCardInfo={this.state.creditCardInfo}
            submitInfo={this.submitInfo.bind(this)}/>
   }
   if(route.name == 'ReviewOrder') {
     return <ReviewOrder navigator={navigator}
            userInfo={this.state.userInfo}
            bookToOrder={this.state.bookToOrder}
            creditCardInfo={this.state.creditCardInfo}
            useDeliveryAddressAsBilling={this.state.useDeliveryAddressAsBilling}
            billingAddress={this.state.billingAddress}
            deliveryAddress={this.state.deliveryAddress}
            submitInfo={this.submitInfo.bind(this)}/>
   }
  }
  render() {
   return (
     <View style={styles.container}>
       <Image source={require('./images/books.png')}
         style={styles.books}>
         <View style={styles.container}>
           <Navigator
             initialRoute={{name: 'HomeScreen'}}
             renderScene={this.renderScene.bind(this)}
           />
         </View>
       </Image>
     </View>
   );
  }
}

function mapStateToProps(state) {
  return {
    deliveryAddress: state.deliveryAddress,
    billingAddress: state.billingAddress
  }
}

function mapDispatchToProps(dispatch) {
  return {
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Root);

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  books: {
    flex: 1,
    width: null,
    resizeMode: 'cover'
  },
});
