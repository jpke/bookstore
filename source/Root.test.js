import React from 'react';
import ReactNative from 'react-native';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from './store';
const store = configureStore();

import Root from './Root';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Root />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
