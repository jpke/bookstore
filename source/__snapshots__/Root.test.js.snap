// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "flex": 1,
        "resizeMode": "cover",
        "width": null,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
              "overflow": "hidden",
            }
          }
        >
          <View
            collapsable={false}
            onStartShouldSetResponderCapture={[Function]}
            pointerEvents="auto"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scaleX": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                    Object {
                      "rotate": "0deg",
                    },
                    Object {
                      "skewX": "0deg",
                    },
                    Object {
                      "skewY": "0deg",
                    },
                  ],
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "space-around",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 3,
                    "marginBottom": 10,
                    "marginTop": 15,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "color": "#ffcd67",
                      "flex": 1,
                      "fontSize": 40,
                      "textShadowColor": "black",
                      "textShadowOffset": Object {
                        "height": 1,
                        "width": 1,
                      },
                      "textShadowRadius": 4,
                    }
                  }
                >
                  Pocket BookStore
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "flex": 1,
                      "fontSize": 30,
                      "textAlign": "center",
                      "textShadowColor": "black",
                      "textShadowOffset": Object {
                        "height": 1,
                        "width": 1,
                      },
                      "textShadowRadius": 4,
                    }
                  }
                >
                  Select from Available Books
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 6,
                    "width": 320,
                  }
                }
              >
                <ScrollView
                  style={
                    Object {
                      "flex": 6,
                      "width": 320,
                    }
                  }
                >
                  <View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "backgroundColor": "rgba(255,205,103,0.8)",
                            "borderRadius": 5,
                            "flex": 1,
                            "marginBottom": 6,
                            "marginTop": 6,
                            "padding": 6,
                          },
                        ]
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <View>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#fff",
                              "fontSize": 22,
                              "padding": 5,
                              "textShadowColor": "black",
                              "textShadowOffset": Object {
                                "height": 1,
                                "width": 1,
                              },
                              "textShadowRadius": 4,
                            }
                          }
                        >
                          The Evolution of Civilizations
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 18,
                              "padding": 5,
                            }
                          }
                        >
                          Carroll Quigley
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#dddddd",
                          "height": 1,
                        }
                      }
                    />
                  </View>
                  <View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "backgroundColor": "rgba(255,205,103,0.8)",
                            "borderRadius": 5,
                            "flex": 1,
                            "marginBottom": 6,
                            "marginTop": 6,
                            "padding": 6,
                          },
                        ]
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <View>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#fff",
                              "fontSize": 22,
                              "padding": 5,
                              "textShadowColor": "black",
                              "textShadowOffset": Object {
                                "height": 1,
                                "width": 1,
                              },
                              "textShadowRadius": 4,
                            }
                          }
                        >
                          How the Mind Works
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 18,
                              "padding": 5,
                            }
                          }
                        >
                          Stephan Pinker
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#dddddd",
                          "height": 1,
                        }
                      }
                    />
                  </View>
                </ScrollView>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1.5,
                    "flexDirection": "row",
                    "justifyContent": "space-around",
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#6799FF",
                        "borderRadius": 10,
                        "margin": 5,
                        "padding": 10,
                        "width": 100,
                      },
                    ]
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                      }
                    }
                  >
                    Orders
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#6799FF",
                        "borderRadius": 10,
                        "margin": 5,
                        "padding": 10,
                        "width": 100,
                      },
                    ]
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                      }
                    }
                  >
                    Admin
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Image>
</View>
`;
